File: mobile/src/main/kotlin/de/scout/fireplace/inject/ApplicationComponent.java
package de.scout.fireplace.inject;

import dagger.Component;
import dagger.android.AndroidInjector;
import dagger.android.support.AndroidSupportInjectionModule;
import de.scout.fireplace.Fireplace;
import de.scout.fireplace.firebase.FirebaseModule;
import de.scout.fireplace.network.NetworkModule;

@Component(
    modules = {
        ApplicationModule.class,
        AndroidSupportInjectionModule.class,
        ActivityBindingModule.class,
        FirebaseModule.class,
        NetworkModule.class,
    }
)
interface ApplicationComponent extends AndroidInjector<Fireplace> {
}

File: feature/src/main/kotlin/de/scout/fireplace/feature/inject/ApplicationComponent.kt
package de.scout.fireplace.feature.inject

import android.app.Application
import dagger.BindsInstance
import dagger.Component
import dagger.android.AndroidInjector
import dagger.android.support.AndroidSupportInjectionModule
import de.scout.fireplace.feature.Fireplace
import de.scout.fireplace.feature.firebase.FirebaseModule
import de.scout.fireplace.feature.network.NetworkModule
import javax.inject.Singleton

@ApplicationScope
@Component(modules = arrayOf(
    ApplicationModule::class,
    AndroidSupportInjectionModule::class,
    ActivityBindingModule::class,
    FirebaseModule::class,
    NetworkModule::class,
    ViewModelModule::class
))
internal interface ApplicationComponent : AndroidInjector<Fireplace> {

  @Component.Builder
  interface Builder {

    @BindsInstance
    fun application(application: Application): Builder

    fun build(): ApplicationComponent
  }
}
